### PR TITLE
Fix home when modifying username

### PR DESCRIPTION
**Pull request by maresb**
In the `usermod` command, I mistakenly set the destination home directory to the old username instead of the new username.

A failing test has been added, then bugfix so that the test now passes.

Output of test before bugfix:
```text
 ✗ docker run --rm -e MAMBA_USER=MaMbAmIcRo micromamba:test-debian-bullseye-slim-modify-username bash -c "realpath ~MaMbAmIcRo"
   (from function `assert_output' in file test/test_helper/bats-assert/src/assert_output.bash, line 194,
    in test file test/user.bats, line 93)
     `assert_output "/home/$altered_mamba_user"' failed
   
   -- output differs --
   expected : /home/MaMbAmIcRo
   actual   : /home/mambauser
   --
```

### CODE REVIEW

**Comment by maresb** [comment]
Perhaps I should point out that the bug is in the documentation and not in the build. The image will not change.

File: test/user.bats [hunk 1]
```diff
@@ -86,6 +86,13 @@ setup() {
         assert_output "$altered_mamba_user"
 }
 
+# Test that home is moved when modifying the username.
+@test "docker run --rm -e MAMBA_USER=$altered_mamba_user ${MICROMAMBA_IMAGE}-modify-username bash -c \"realpath ~${altered_mamba_user}\"" {
+        run docker run --rm -e MAMBA_USER=$altered_mamba_user ${MICROMAMBA_IMAGE}-modify-username bash -c "realpath ~${altered_mamba_user}"
```
Comment (wholtz) [inline test/user.bats:91]: Add quotes to guard against  variables containing spaces:

```suggestion
        run docker run --rm -e "MAMBA_USER=$altered_mamba_user" "${MICROMAMBA_IMAGE}-modify-username" bash -c "realpath ~${altered_mamba_user}"
```

**Review by wholtz — CHANGES_REQUESTED** [review]
Thanks for catching this and doing the update!

File: test/user.bats (hunk 1 above)
Comment (maresb) [reply test/user.bats]: Thanks. For my information, this is _purely_ on principle, right? Firstly, valid usernames and image names cannot contain spaces. Secondly this is for tests where we control the inputs.

It's awkward to escape the quotes in the `@test` line.

**Comment by wholtz** [comment]
Did you delete your comment on spaces in usernames and image names? I saw the notification on my phone and then opened my laptop to respond and now I can't find it.

I'm not sure if spaces are technically allowed in unix usernames or docker images. Or if there is an edge case using unicode characters. I prefer to follow the normal quoting rules rather than to try to dig into these details.

**Review by wholtz — APPROVED** [review]

**Comment by maresb** [comment]
@wholtz, got it, thanks!

I made my comment under your commit suggestion which has since been collapsed, which is probably why you couldn't find it. (I probably should have asked in the main thread.)

File: test/user.bats (hunk 1 above)
Comment (wholtz) [reply test/user.bats]: Okay, I manged to find your comment again. Where you put it is fine, I just didn't realize that review comments get collapsed within the PR conversation. I've been trying to make use of the Github review functionality more, but obviously I'm still learning some of the details.

**Pull request merged**